pandas/core/strings.py

```diff
@@ -939,7 +942,7 @@ def str_extractall(arr, pat, flags=0):
     if regex.groups == 0:
         raise ValueError("pattern contains no capture groups")
 
-    if isinstance(arr, ABCIndex):
+    if isinstance(arr, ABCIndexClass):
```

Comment:
This line fixes #23555.

Reply:
of course, ABCIndexClass is the base index class.